### ISSUE TITLE
RFD: GUI Redesgin

### ISSUE THREAD

**Issue by rkitover** [visualboyadvance-m/visualboyadvance-m#504]
@denisfa @ZachBacon 

So we've been planning for a while to do an overhaul of the interface, mostly the settings/options, to make something more modern and streamlined. Removing unnecessary/outdated elements etc.

A lot of this would be easy to do, mostly just rearranging the structure of the XRC.

Let's discuss it here.

**Comment by ZachBacon** [comment]
I completely agree, things like the preferences I think sound be a central location instead of having their own separate dialogues. 

Something along say hexchat (but obviously it's gonna be much different than hexchat's but just to give an idea) 
![preferences](https://user-images.githubusercontent.com/8948676/64909597-665c8780-d6dc-11e9-9b1d-50f1a4f9b4a8.png)

There's obviously other things that can be changed. But I definitely propose a central location for preferences.

**Comment by denisfa** [comment]
I suggest to create/destroy the window instead of just hiding/showing when you press the menu. I don't know much about other eventual issues of the current gui, so can't talk about that.

Also, is this the case of keeping the wxWidgets `2.8`? I would also want to update it to at least `3.0` (I personally want the latest, `3.1.2`).

**Comment by ZachBacon** [comment]
I suggest at potentially doing some design mockups, see if we can get some form of conformant design going, of how we want a modernized vba-m should look dialogue wise.

**Comment by mudlord** [comment]
Modernized should have that hexchat like dialog.

**Comment by ZachBacon** [comment]
Well I only suggested that @mudlord due to the fact it has all the settings in one place instead of having to go to various menu clicks to try and find the right prompt you want. It was just an example that's all.

**Comment by mudlord** [comment]
well for some settings a more refined experience for instance using checkboxes in the gui. atm, having a all encompassing setting for system paths, input, would be nice.
